webpack/scenes/ContentViews/Details/Filters/MatchContentModal/__tests__/CVRpmMatchContentModal.test.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import { renderWithRedux, patientlyWaitFor, fireEvent } from 'react-testing-lib-wrapper';
+
+
+import api from '../../../../../../services/api';
+import CVRpmMatchContentModal from '../CVRpmMatchContentModal';
+import nock, { nockInstance, assertNockRequest, mockSetting, mockAutocomplete } from '../../../../../../test-utils/nockWrapper';
+
+const CVMatchedContent = require('./CVRpmMatchContent.fixtures.json');
+const CVMatchContentSearch = require('./CVRpmMatchContentSearch.fixtures.json');
+
+const firstMatchContent = CVMatchedContent.results[0];
+const { nvra: secondMatchContentName } = CVMatchedContent.results[1];
+const cvMatchContentPath = api.getApiUrl('/packages');
+const autocompleteUrl = '/packages/auto_complete_search';
+
+const onClose = jest.fn();
+
+const MatchContentModal =
+  <CVRpmMatchContentModal filterId={194} filterRuleId={13} onClose={onClose} />;
+
+let searchDelayScope;
+let autoSearchScope;
+beforeEach(() => {
+  searchDelayScope = mockSetting(nockInstance, 'autosearch_delay', 500);
+  autoSearchScope = mockSetting(nockInstance, 'autosearch_while_typing', true);
+});
+
+afterEach(() => {
+  nock.cleanAll();
+  assertNockRequest(searchDelayScope);
+  assertNockRequest(autoSearchScope);
+});
+
+
+test('Can show matching content modal', async (done) => {
+  const autocompleteScope = mockAutocomplete(nockInstance, autocompleteUrl);
+  const createMatchContentscope = nockInstance
+    .get(cvMatchContentPath)
+    .query(true)
+    .reply(200, CVMatchedContent);
+
+  const { queryByText } = renderWithRedux(MatchContentModal);
+  await patientlyWaitFor(() => {
+    expect(queryByText('Matching content')).toBeInTheDocument();
+    expect(queryByText(firstMatchContent.nvra)).toBeInTheDocument();
+  });
+
+  assertNockRequest(autocompleteScope);
+  assertNockRequest(createMatchContentscope, done);
+});
+
+test('Can search with filter', async (done) => {
+  const autocompleteScope = mockAutocomplete(nockInstance, autocompleteUrl);
+  const getMatchContentscope = nockInstance
+    .get(cvMatchContentPath)
+    .query(true)
+    .reply(200, CVMatchedContent);
+
+  const autocompleteSearchScope = nockInstance
+    .get(api.getApiUrl(autocompleteUrl))
+    .query(true)
+    .reply(200, []);
+
+  const secondGetMatchContentscope = nockInstance
+    .get(cvMatchContentPath)
+    .query(true)
+    .reply(200, CVMatchContentSearch);
+
+  const { queryByText, getByLabelText } = renderWithRedux(MatchContentModal);
+
+  await patientlyWaitFor(() => {
+    expect(queryByText('Matching content')).toBeInTheDocument();
+    expect(queryByText(secondMatchContentName)).toBeInTheDocument();
+  });
+
+  fireEvent.change(getByLabelText(/text input for search/i), { target: { value: firstMatchContent.nvra } });
+
+  await patientlyWaitFor(() => {
+    expect(queryByText('Matching content')).toBeInTheDocument();
+    expect(queryByText(secondMatchContentName)).not.toBeInTheDocument();
```

Comment:
For completeness, also expect firstMatchContent to be on page?

Reply:
sure!